x-pack/plugins/maps/public/components/widget_overlay/layer_control/layer_toc/toc_entry/view.js

```diff
@@ -152,78 +201,78 @@ export class TOCEntry extends React.Component {
       fitToBounds
     } = this.props;
 
-    let sortIcon;
-    if (!isReadOnly) {
-      sortIcon = (
-        <EuiFlexItem grow={false}>
-          <span className="mapTocEntry__grab" {...this.props.dragHandleProps}>
-            <EuiIcon type="grab"/>
-          </span>
-        </EuiFlexItem>
-      );
-    }
-
     return (
-      <EuiFlexGroup
-        gutterSize="none"
-        alignItems="center"
-        responsive={false}
+      <div
         className={
           layer.isVisible() && layer.showAtZoomLevel(zoom)
             && !layer.hasErrors() ? 'mapTocEntry-visible' : 'mapTocEntry-notVisible'
         }
       >
-        <EuiFlexItem grow={false}>
-          <LayerTocActions
-            layer={layer}
-            fitToBounds={() => {
-              fitToBounds(layer.getId());
-            }}
-            zoom={zoom}
-            toggleVisible={() => {
-              toggleVisible(layer.getId());
-            }}
-            displayName={this.state.displayName}
-            cloneLayer={() => {
-              cloneLayer(layer.getId());
-            }}
-            isReadOnly={this.props.isReadOnly}
-          />
-        </EuiFlexItem>
-        <EuiFlexItem>
-          {this._renderLayerName()}
-        </EuiFlexItem>
-        {sortIcon}
-      </EuiFlexGroup>
+        <LayerTocActions
+          layer={layer}
+          fitToBounds={() => {
+            fitToBounds(layer.getId());
+          }}
+          zoom={zoom}
+          toggleVisible={() => {
+            toggleVisible(layer.getId());
+          }}
+          displayName={this.state.displayName}
+          escapedDisplayName={escapeLayerName(this.state.displayName)}
+          cloneLayer={() => {
+            cloneLayer(layer.getId());
+          }}
+          editLayer={this._openLayerPanelWithCheck}
+          isReadOnly={isReadOnly}
+        />
+
+        {this._renderLayerIcons()}
+      </div>
     );
   }
 
-  _renderLayerDetails() {
+  _renderLayerDetails = () => {
```

Comment:
consider renaming `layerDetails` to `legendDetails` (similar for isLayerDetailsOpen). It's starting to get a little confusing (layer-settings panel, layer-detail in layer-settings, layer-details that are actually legend-details, ...